.github/workflows/ci.yml

```diff
@@ -40,6 +40,11 @@ jobs:
             calculate-code-coverage: 'yes'
             phpunit-flags: '--testsuite coverage --exclude-group covers-nothing --coverage-clover build/logs/clover.xml'
 
+          - operating-system: 'ubuntu-20.04'
+            php-version: '7.4'
+            job-description: 'with deployment'
+            execute-deployment: 'yes'
```

Comment:
note: it actually does not do deployment itself. but it run the smokes tests, so we can make sure that PR won't break the main branch